be/src/exec/pipeline/scan/morsel.h

```diff
@@ -82,14 +82,16 @@ class Morsel {
     void set_from_version(int64_t from_version) { _from_version = from_version; }
     int64_t from_version() { return _from_version; }
 
-    void set_rowsets(std::vector<RowsetSharedPtr> rowsets) { _rowsets = std::move(rowsets); }
-    const std::vector<RowsetSharedPtr>& rowsets() const { return _rowsets; }
+    void set_rowsets(const std::vector<RowsetSharedPtr>& rowsets) { _rowsets = &rowsets; }
```

Comment:
why we change move object to acquire pointer?
It also looks like move has no additional overhead.

Reply:
Multiple morsels uses the same `_rowsets` from `MorselQueue::_rowsets_per_tablet`, so we cannot use `set_rowsets(std::move(_rowsets_per_tablet[i]))` but `set_rowsets(_rowsets_per_tablet[i])`.
Therefore, if we use move object, we still need copy from the actual parameter to the formal parameter.

Reply:
got it thanks.